### ISSUE TITLE
get_cache doesn't actually get the cache at the path provided

### ISSUE THREAD

**Issue by kendonB** [ropensci/drake#883]
## Prework

- [x] Read and abide by `drake`'s [code of conduct](https://github.com/ropensci/drake/blob/master/CODE_OF_CONDUCT.md).
- [x] Search for duplicates among the [existing issues](https://github.com/ropensci/drake/issues), both open and closed.
- [x] Advanced users: verify that the bug still persists in the current development version (i.e. `remotes::install_github("ropensci/drake")`) and mention the [SHA-1 hash](https://git-scm.com/book/en/v1/Getting-Started-Git-Basics#Git-Has-Integrity) of the [Git commit you install](https://github.com/ropensci/drake/commits/master).

## Description

`get_cache` finds `.drake` even when the exact path of an adjacent cache is given.

## Reproducible example

``` r
library(drake)
cache = new_cache(".test")
test_plan = drake_plan(
  dot_test = 1
)
make(test_plan, cache = cache)
#> target dot_test
dr_plan = drake_plan(
  dot_drake = 1
)
make(dr_plan)
#> target dot_drake
# finds .drake
cached(cache = get_cache(".test"))
#> [1] "dot_drake"
# works
cached(cache = cache)
#> [1] "dot_test"
# works
cached(cache = this_cache(".test"))
#> Warning: this_cache() in drake is deprecated. Use get_cache() or
#> storr::storr_rds() instead.
#> [1] "dot_test"
```

<sup>Created on 2019-05-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

## Expected output

``` r
library(drake)
cache = new_cache(".test")
test_plan = drake_plan(
  dot_test = 1
)
make(test_plan, cache = cache)
#> target dot_test
dr_plan = drake_plan(
  dot_drake = 1
)
make(dr_plan)
#> target dot_drake
# works
cached(cache = get_cache(".test"))
#> [1] "dot_test"
# works
cached(cache = cache)
#> [1] "dot_test"
# works
cached(cache = this_cache(".test"))
#> Warning: this_cache() in drake is deprecated. Use get_cache() or
#> storr::storr_rds() instead.
#> [1] "dot_test"
```

<sup>Created on 2019-05-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by wlandau** [comment]
Thanks for the nudge. I regret the awkward design of `get_cache()`, and I think we should deprecate `get_cache()` in favor of a better-behaved `drake_cache()` function. A PR is forthcoming.

**Comment by wlandau** [comment]
Fixed via #884.
